pkg/controller/config/config_controller.go

```diff
@@ -249,14 +251,22 @@ func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result
 
 	// If the watch set has not changed, we're done here.
 	if r.watched.Equals(newSyncOnly) && r.processExcluder.Equals(newExcluder) {
-		return reconcile.Result{}, nil
+		if !r.wiped && r.needsReplay != nil {
```

Comment:
I _think_ this condition still isn't quite right 🤔. Submitting a patch to your patch that hopefully improves clarity a bit. Let me know what you think.

Reply:
https://github.com/maxsmythe/gatekeeper/pull/1

Reply:
Yep, was definitely still broken. I took a look at your suggestions. I think they're mostly great. 1 question regarding thread safety on the changes.

Reply:
merged suggestions, thanks again!